sdk/src/main/java/io/dapr/client/domain/State.java

```diff
@@ -191,6 +230,10 @@ public boolean equals(Object o) {
       return false;
     }
 
+    if (getMetadata() != null ? !getMetadata().equals(that.getMetadata()) : that.getMetadata() != null) {
```

Comment:
Does this equal method work for maps? Will it make a deep comparison? This might be worth unit testing. Can you add unit test for this class to validate equals and hashcode?